src/status/wallet.nim

```diff
@@ -50,6 +71,36 @@ proc initEvents*(self: WalletModel) =
 proc delete*(self: WalletModel) =
   discard
 
+proc trackPendingTransaction*(self: WalletModel, address: string, trxHash: string, trxType: PendingTransactionType, data: string) =
+  let latestBlock = getBlockByNumber("latest").parseJson()["result"].getInt
+
+  if not self.pendingTransactions.hasKey(address):
+    self.pendingTransactions[address] = @[]
+
+  self.pendingTransactions[address].add PendingTransaction(
+    transactionHash: trxHash,
+    trxType: trxType,
+    blockNumber: latestBlock,
+    data: data,
+    mined: false
+  )
+
+proc getTransactionReceipt*(self: WalletModel, transactionHash: string): JsonNode =
+  result = status_wallet.getTransactionReceipt(transactionHash).parseJSON()["result"]
+
+proc checkPendingTransactions*(self: WalletModel, address: string, blockNumber: int) =
+  if not self.pendingTransactions.hasKey(address): return
+  for trx in self.pendingTransactions[address].mitems:
+    if trx.mined: continue
+
+    let transactionReceipt = self.getTransactionReceipt(trx.transactionHash)
+    if transactionReceipt.kind != JNull:
+      trx.mined = true
```

Comment:
Should we remove the `PendingTransaction` from memory? Or do we still need it?

Reply:
I changed the PendingTransactions to handle them in status-go instead of doing it in desktop in this PR, so after processing the transaction I remove it from the DB: #880